pkg/alertmanager/statefulset.go

```diff
@@ -531,8 +534,14 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config, tlsAssetSe
 	amCfg := a.Spec.AlertmanagerConfiguration
 	if amCfg != nil && len(amCfg.Templates) > 0 {
 		sources := []v1.VolumeProjection{}
+		configmapKeys := sets.Set[string]{}
+		secretKeys := sets.Set[string]{}
```

Comment:
If a secret and a configmap have the same name, I believe we'll also have conflicting paths, is that correct?

Don't we need to use the same set for both?

Reply:
Hmm, that makes sense. Patched!